lib/archive/util.go

```diff
@@ -106,25 +120,26 @@ func AddMountInclusionsExclusions(currentMount string, filter *FilterSpec, mount
 	for _, mount := range mounts {
 		if strings.HasPrefix(mount, currentMount) && currentMount != mount {
 			// exclusions are relative to the mount so the leading `/` should be removed unless we decide otherwise.
-			exclusion := removeLeadingSlash(strings.TrimPrefix(mount, currentMount))
-			if !strings.HasSuffix(exclusion, "/") {
-				exclusion = exclusion + "/"
-			}
-
+			exclusion := clean(strings.TrimPrefix(mount, currentMount), true)
+			exclusion = strings.TrimPrefix(exclusion, "/") + "/"
```

Comment:
don't know that this second TrimPrefix is needed given the call to `clean(..., true)`